### PR TITLE
[IOTDB-3428] Linear expanse RegionGroup(simple version)

### PR DESCRIPTION
**Pull request by CRZbulabula**
The following two scenarios trigger RegionGroup expansion:

1. When a StorageGroup writes schema/data for the first time, initialize a SchemaRegionGroup/DataRegionGroup for it.
2. When a StorageGroup writes schema/data continuously, extend some SchemaRegionGroups/DataRegionGroups until its _slotCount\regionCount <= maxSlotCount\maxRegionCount_.

The following two scenarios trigger the adjustment of maxRegionGroupCount:

1. Register/Remove DataNode.
2. Set/Delete StorageGroup.

See [maxRegionGroupCount](https://apache-iotdb.feishu.cn/docs/doccnHuDkDD6RiwLjaej3hkA9Vb#whSIka) for detailed adjustment formulas.

### CODE REVIEW

**Comment by CRZbulabula** [comment]
WIP.

**Comment by CRZbulabula** [comment]
Ready for review~

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/ClusterSchemaManager.java [hunk 1]
```diff
@@ -133,31 +140,113 @@ public TSStatus setTimePartitionInterval(
   }
 
   /**
-   * Count StorageGroups by specific path pattern
+   * Only leader use this interface. Adjust the maxSchemaRegionGroupCount and
+   * maxDataRegionGroupCount of each StorageGroup bases on existing cluster resources
+   */
+  public void adjustMaxRegionGroupCount() {
```
[the remark this answers is not in the document]
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/ClusterSchemaManager.java]: Fixed

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**